File: MovieCatalogueV2/app/src/main/java/com/google/fdp/moviecataloguev2/networks/MovieService.java
package com.google.fdp.moviecataloguev2.networks;

import com.google.fdp.moviecataloguev2.models.Movie;
import com.google.fdp.moviecataloguev2.models.responses.BaseResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by gama on 2019-07-26.
 * Addin Gama Bertaqwa
 * addingama@gmail.com
 */
public interface MovieService {
    @GET("3/discover/movie")
    Call<BaseResponse<Movie>> getMovies(@Query("api_key") String apiKey);

    @GET("3/search/movie?language=en-US")
    Call<BaseResponse<Movie>> searchMovies(@Query("api_key") String apiKey, @Query("query") String keyword);

    @GET("3/discover/tv")
    Call<BaseResponse<Movie>> getTvSeries(@Query("api_key") String apiKey);

    @GET("3/search/tv?language=en-US")
    Call<BaseResponse<Movie>> searchTvSeries(@Query("api_key") String apiKey, @Query("query") String keyword);

    @GET("3/discover/movie")
    Call<BaseResponse<Movie>> getMovieReleaseToday(@Query("api_key") String apiKey,
                                                   @Query("primary_release_date.gte") String startDate,
                                                   @Query("primary_release_date.lte") String endDate);

}

File: MovieCatalog 3/app/src/main/java/com/alfianh/moviecatalog/network/MovieService.kt
package com.alfianh.moviecatalog.network

import com.alfianh.moviecatalog.models.Movie
import com.alfianh.moviecatalog.models.response.BaseResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface MovieService {

  @GET("discover/movie")
  fun getMovies(): Call<BaseResponse<Movie>>

  @GET("discover/tv")
  fun getTvShows(): Call<BaseResponse<Movie>>

  @GET("search/movie")
  fun searchMovies(@Query("query")
  query: String): Call<BaseResponse<Movie>>

}